wom/src/main/scala/wom/values/WomArray.scala

```diff
@@ -59,7 +62,16 @@ sealed abstract case class WomArray(womType: WomArrayType, value: Seq[WomValue])
       case _ => this
     }
   }
-  
+
+  def traverse[R <: WomValue, G[_]](f: WomValue => G[R])(implicit applicative: Applicative[G]): G[WomArray] = {
+    if (value.isEmpty) applicative.pure(this)
+    else {
+      applicative.map(value.toList.traverse(f)) { mapped =>
+        WomArray(WomArrayType(mapped.head.womType), mapped)
+      }
+    }
+  }
```

Comment:
oh you mean like writing a `Traverse[WomArray]` ? That'd be cool I'll give it a shot
For the type I copied what `map` is doing just above but I suspect you're right

Reply:
Unfortunately it doesn't seem possible now because traversable WomTypes are not truly generic. If `WomArray(WomArrayType(WomStringType), List(WomString("hello"))` was instead `WomArray[WomString](WomString("hello"))` I think we could make an instance of `Traverse[WomArray]` and use cats.traverse directly on a `WomArray` value.